stream/Cargo.toml

```diff
@@ -20,7 +20,9 @@ thiserror = { workspace = true }
 bytes = { workspace = true }
 futures = { workspace = true }
 rand = { workspace = true }
+sha2 = { workspace = true }
 chacha20poly1305 = "0.10"
```

Comment:
You didn't introduce this but can you ensure all versions here are fully specified (x.y.z)?